spring-kafka/src/test/java/org/springframework/kafka/support/DefaultKafkaHeaderMapperTests.java

```diff
@@ -332,6 +334,115 @@ void inboundJson() {
+		rawHeaders.add(multiValueWildeCardHeader2, new byte[] { 0, 0, 0, 9 });
+
+		// WHEN
+		Map<String, Object> mappedHeaders = new HashMap<>();
+		mapper.toHeaders(rawHeaders, mappedHeaders);
+
+		// THEN
+		assertThat(mappedHeaders.get(KafkaHeaders.DELIVERY_ATTEMPT)).isEqualTo(1);
+		assertThat(mappedHeaders.get(KafkaHeaders.ORIGINAL_OFFSET)).isEqualTo(new byte[] { 0, 0, 0, 1 });
+		assertThat(mappedHeaders.get(RetryTopicHeaders.DEFAULT_HEADER_ATTEMPTS)).isEqualTo(new byte[] { 0, 0, 0, 5 });
+		assertThat(mappedHeaders.get(singleValueHeader)).isEqualTo(new byte[] { 0, 0, 0, 6 });
+
+		@SuppressWarnings("unchecked")
+		List<byte[]> multiHeader1Values = (List<byte[]>) mappedHeaders.get(multiValueHeader1);
+		assertThat(multiHeader1Values).contains(new byte[] { 0, 0, 0, 0 },
+												new byte[] { 0, 0, 0, 1 },
+												new byte[] { 0, 0, 0, 2 },
+												new byte[] { 0, 0, 0, 3 });
+
+		@SuppressWarnings("unchecked")
+		List<byte[]> multiHeader2Values = (List<byte[]>) mappedHeaders.get(multiValueHeader2);
+		assertThat(multiHeader2Values).contains(new byte[] { 0, 0, 0, 4 },
+												new byte[] { 0, 0, 0, 5 });
+
+		@SuppressWarnings("unchecked")
+		List<byte[]> multiValueWildeCardHeader1Values = (List<byte[]>) mappedHeaders.get(multiValueWildeCardHeader1);
+		assertThat(multiValueWildeCardHeader1Values).contains(new byte[] { 0, 0, 0, 6 },
+															new byte[] { 0, 0, 0, 7 });
+
+		@SuppressWarnings("unchecked")
+		List<byte[]> multiValueWildeCardHeader2Values = (List<byte[]>) mappedHeaders.get(multiValueWildeCardHeader2);
+		assertThat(multiValueWildeCardHeader2Values).contains(new byte[] { 0, 0, 0, 8 },
+															new byte[] { 0, 0, 0, 9 });
+	}
+
+	@Test
+	void multiValueHeaderFromTest() {
+		// GIVEN
+		String multiValueHeader1 = "test-multi-value1";
+		String multiValueHeader2 = "test-multi-value2";
+		String multiValueHeader3 = "test-other-multi-value1";
+		String multiValueHeader4 = "test-prefix-match-multi";
+		String singleValueHeader = "test-single-value1";
+
+		Message<String> message = MessageBuilder
+				.withPayload("test-multi-value-header")
+				.setHeader(multiValueHeader1, List.of(new byte[] { 0, 0, 0, 1 },
+													new byte[] { 0, 0, 0, 2 }))
+				.setHeader(multiValueHeader2, List.of(new byte[] { 0, 0, 0, 3 },
+													new byte[] { 0, 0, 0, 4 }))
+				.setHeader(multiValueHeader3, List.of(new byte[] { 0, 0, 0, 9 },
+													new byte[] { 0, 0, 0, 10 }))
+				.setHeader(multiValueHeader4, List.of(new byte[] { 0, 0, 0, 11 },
+													new byte[] { 0, 0, 0, 12 }))
+				.setHeader(singleValueHeader, new byte[] { 0, 0, 0, 5 })
+				.build();
+
+		DefaultKafkaHeaderMapper mapper = new DefaultKafkaHeaderMapper();
+		mapper.setMultiValueHeaderPatterns("test-multi-*",
+											multiValueHeader3,
+											"*-prefix-match-multi*");
+
+		// WHEN
+		Headers results = new RecordHeaders();
+		mapper.fromHeaders(message.getHeaders(), results);
+
+		// THEN
+		assertThat(results).contains(
```

Comment:
I think we can utilize `RecordHeaders` API for much cleaner assertion. This `contains` is just not readable.